File: apps/Applications/ios/Local For Vocal/Local For Vocal/Sources/Views/Pages/Grocery/Components/GroceryWholesaleTextComponent.swift
import SwiftUI

struct GroceryWholesaleTextComponent: View {
    let component: SDUIComponent
    @StateObject private var viewModel = SDUIComponentViewModel()

    var body: some View {
        VStack(spacing: 4) {
            if let lines = viewModel.data as? [WholesaleTextLine] {
                ForEach(lines, id: \.self) { line in
                    Text(line.text)
                        .font(.system(size: line.size, weight: line.fontWeight))
                        .foregroundColor(Color(hex: line.color))
                        .multilineTextAlignment(.center)
                        .fixedSize(horizontal: false, vertical: true)
                }
            } else {
                // Fallback / Loading state
                Text("You Just Experienced")
                    .font(.system(size: 20, weight: .medium))
                    .foregroundColor(Color(hex: "#D4AF37"))  // Goldish
                Text("3000+")
                    .font(.system(size: 56, weight: .heavy))
                    .foregroundColor(Color(hex: "#D4AF37"))
                Text("Products at\nWholesale Prices")
                    .font(.system(size: 34, weight: .bold))
                    .foregroundColor(Color(hex: "#D4AF37"))
                    .multilineTextAlignment(.center)
            }
        }
        .frame(maxWidth: .infinity)
        .padding(.top, 50)
        .padding(.bottom, 10)
        .padding(.horizontal, 16)
        .background(Color.white)  // Or transparent
        .onAppear {
            viewModel.decodeItems(from: component, type: WholesaleTextLine.self, key: "lines")
        }
    }
}

struct WholesaleTextLine: Codable, Hashable {
    let text: String
    let size: CGFloat
    let color: String
    let weight: String?  // "bold", "medium", "regular", "heavy"

    var fontWeight: Font.Weight {
        switch weight?.lowercased() {
        case "ultraLight": return .ultraLight
        case "thin": return .thin
        case "light": return .light
        case "regular": return .regular
        case "medium": return .medium
        case "semibold": return .semibold
        case "bold": return .bold
        case "heavy": return .heavy
        case "black": return .black
        default: return .bold
        }
    }
}

File: apps/Applications/android/app/src/main/java/com/ecommerceearn/app/ui/components/GroceryWholesaleTextComponent.kt
package com.ecommerceearn.app.ui.components

import androidx.compose.foundation.layout.*
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.ecommerceearn.app.data.model.SDUIComponent
import com.google.gson.Gson

@Composable
fun GroceryWholesaleTextComponent(component: SDUIComponent) {
    val props = component.props

    data class WholesaleTextLine(val text: String, val size: Float, val color: String, val weight: String?)
    
    val linesArr = props?.getArray("lines")
    val lines = linesArr?.mapNotNull {
        try { Gson().fromJson(it, WholesaleTextLine::class.java) } catch(e:Exception){ null }
    } ?: emptyList()

    Column(
        modifier = Modifier
            .fillMaxWidth()
            .padding(top = 50.dp, bottom = 10.dp, start = 16.dp, end = 16.dp),
        horizontalAlignment = Alignment.CenterHorizontally
    ) {
        if (lines.isNotEmpty()) {
            lines.forEach { line ->
                val fontWeight = when (line.weight?.lowercase()) {
                    "bold" -> FontWeight.Bold
                    "medium" -> FontWeight.Medium
                    "heavy" -> FontWeight.Black
                    "light" -> FontWeight.Light
                    else -> FontWeight.Bold
                }
                Text(
                    text = line.text,
                    fontSize = line.size.sp,
                    fontWeight = fontWeight,
                    color = safeParseColor(line.color),
                    textAlign = TextAlign.Center
                )
            }
        } else {
            // Fallback identical to iOS
            Text("You Just Experienced", fontSize = 20.sp, fontWeight = FontWeight.Medium, color = Color(0xFFD4AF37), textAlign = TextAlign.Center)
            Text("3000+", fontSize = 56.sp, fontWeight = FontWeight.Black, color = Color(0xFFD4AF37), textAlign = TextAlign.Center)
            Text("Products at\nWholesale Prices", fontSize = 34.sp, fontWeight = FontWeight.Bold, color = Color(0xFFD4AF37), textAlign = TextAlign.Center)
        }
    }
}
